_posts/2019-06-25-dynamic_linear_models_tfprobability/dynamic_linear_models_tfprobability.Rmd

```diff
@@ -0,0 +1,805 @@
+
+In practical applications, two characteristics of SSMs are especially attractive. 
+
+For one, they let us estimate dynamically changing parameters. In regression, the parameters can be viewed as a hidden state; we may thus have a slope and an intercept that vary over time. When parameters can vary, we speak of _dynamic linear models_ (DLMs). This is the term we'll use throughout this post when referring to this class of models.
+
+Second, linear-Gaussian SSMs are useful in time series forecasting because Gaussian processes can be _added_. A time series can thus be framed as, e.g., the sum of a linear trend and a process that varies seasonally.
+
+Using [tfprobability](https://rstudio.github.io/tfprobability/), the R wrapper to TensorFlow Probability, we illustrate both aspects here. Our first example will be on _dynamic linear regression_. In a detailed walkthrough, we show on how to fit such a model, how to obtain filtered, as well as smoothed, estimates of the coefficients, and how to obtain forecasts. 
+Our second example then illustrates process additivity. This example will build on the first and may also serve as a quick recap of the overall procedure.
+
+Let's jump in. 
+
+
+## Dynamic linear regression example: Capital Asset Pricing Model (CAPM)
+
+Our code builds on the recently released versions of TensorFlow and TensorFlow Probability: 1.14 and 0.7, respectively. 
+
+```{r}
+devtools::install_github("rstudio/tfprobability")
+
+# also installs TensorFlow Probability v. 0.7
+tensorflow::install_tensorflow(version = "1.14")
+
+library(tensorflow)
+library(tfprobability)
+
+library(tidyverse)
+library(zeallot)
+```
+
+
+Note how there's one thing we used to do lately that we're _not_ doing here: We're not enabling eager execution. We say why in a minute.
+
+Our example is taken from Petris et al.(2009)[@Petris-2009], chapter 3.2.7. ^[A draft version is available online at http://people.bordeaux.inria.fr/pierre.delmoral/dynamics-linear-models.petris_et_al.pdf. However, the draft does not yet fully elaborate on this example.]
+Besides introducing the [dlm](https://cran.r-project.org/web/packages/dlm/index.html) package, this book provides a nice introduction to the ideas behind DLMs in general.
+
+To illustrate dynamic linear regression, the authors feature a dataset, originally from Berndt(1991)[@Berndt-1991] that has monthly returns, collected from January 1978 to December 1987, for four different stocks, the 30-day Treasury Bill -- standing in for a _risk-free_ asset --, and the value-weighted average returns for all stocks listed at the New York and American Stock Exchanges, representing the overall _market returns_. 
```

Comment:
There's a lot going on in this sentence. Maybe separate out the variables that the reader needs to know from the data provenance?